### DISCUSSION TITLE
"System.StackOverflowException” in NetVips.dll

### DISCUSSION THREAD

**Discussion by wangmuzhi126** [libvips/libvips#3997] [General]
hi,i miss this proplem "System.StackOverflowException" when i stitch a large image using Netvips.How to deal with it?

**Comment by jcupitt** [comment]
Hello @wangmuzhi126,

libvips will stack overflow with very deep pipelines. I think on windows you can safely have up to about 1,000 nodes, on linux it's about 3,000.

If you've gone over this limit, you will need to restructure your code to make a wider, shallower pipeline.

You'd need to share some code before we could advise on the best way to restructure.

**Reply by feelgood-interface** [reply]
@jcupitt I have a similar issue where I try to generate a thumbnail from a PDF. I am using poppler.
- Using poppler natively works (Windows and Linux). `pdftoppm -f 1 -l 1 -jpeg -singlefile -scale-to 200 plan.pdf thumbnail`
- Using libvips on Linux also works. `vips thumbnail "plan.pdf[page=0]" "thumbnail.jpg" 200`
- Using libvips on Windows doesn't work. `vips thumbnail "plan.pdf[page=0]" "thumbnail.jpg" 200`

Is there something I can configure that could work?

**Reply by jcupitt** [reply]
What's the exact error message?

**Reply by feelgood-interface** [reply]
When I run it in a console app (NetVips) the app crashes (no exception thrown in the code):
`The program '[43540] ConsoleApp1.exe' has exited with code 3221225725 (0xc00000fd) 'Stack overflow'.`

When I use the command line, no info is printed. In the event viewer I have this error:
```
Faulting application name: vips.exe, version: 0.0.0.0, time stamp: 0x65f30ba2
Faulting module name: libpoppler-135.dll, version: 0.0.0.0, time stamp: 0x65f2d43d
Exception code: 0xc00000fd
Fault offset: 0x00000000000f70fc
Faulting process ID: 0x5524
Faulting application start time: 0x01db10c58e32ea34
Faulting application path: C:\dev\vips\bin\vips.exe
Faulting module path: C:\dev\vips\bin\libpoppler-135.dll
Report ID: 275812cf-c68a-4d5c-ab3d-e51bea0cc69d
Faulting package full name: 
Faulting package-relative application ID: 
```

I am using the pre-built libraries from https://github.com/libvips/build-win64-mxe

**Reply by jcupitt** [reply]
I suppose a very complex PDF might trigger a stack overflow. I guess it works for other PDFs you have, is that right?

fwiw, the `vipsthumbnail` (all one word) CLI program is a little more convenient to use, try eg.

```
vipsthumbnail plan.pdf -o thumbnail.jpg --size 200
```

**Reply by feelgood-interface** [reply]
Yes, other PDFs are working fine. With `vipsthumbnail` I get the same error.
If there is nothing I can configure in libvips that could help with this problem, I will try a different PDF library.

**Reply by jcupitt** [reply]
You can configure libvips to use PDFium rather then poppler, but you'd need to rebuild from source.
